packages/web/src/scss/components/Modal/_ModalDialog.scss

```diff
@@ -51,54 +96,19 @@
 }
 
 .ModalDialog {
-    @include typography.generate(theme.$typography);
-
-    --scroll-view-vertical-height: auto; // 2.
-
-    position: fixed;
-    bottom: 0;
-    left: 50%;
-    display: flex;
-    flex-direction: column;
-    box-sizing: border-box; // 3.
-    width: theme.$dialog-width;
-    height: theme.$dialog-height;
-    max-height: theme.$dialog-max-height;
-    overflow-x: hidden;
-    overflow-y: auto;
-    color: theme.$dialog-text-color;
-    border-radius: theme.$dialog-border-radius theme.$dialog-border-radius 0 0;
-    background-color: theme.$dialog-background-color;
-    box-shadow: theme.$dialog-shadow;
-    transform: translate(var(--modal-translate-x), var(--modal-translate-y)) scale(var(--modal-scale));
-    transform-origin: bottom center;
-    overscroll-behavior: contain;
-
-    @include breakpoint.up(map.get(theme.$breakpoints, tablet)) {
-        bottom: 50%;
-        width: theme.$dialog-width-tablet;
-        max-width: calc(100% - #{theme.$padding-x-tablet});
-        height: theme.$dialog-height-tablet;
-        max-height: theme.$dialog-max-height-tablet;
-        border-radius: theme.$dialog-border-radius;
-        transform-origin: center center;
-    }
-
-    @include breakpoint.up(map.get(theme.$breakpoints, desktop)) {
-        width: theme.$dialog-width-desktop;
-    }
+    @include -modal-dialog();
 
     // @deprecated The "uniform" dialog variant is deprecated and will be removed in the next major release.
-    // Migration:
-    // 1. Remove the feature flag and make the uniform dialog the default.
-    // 2. Adjust transitions below.
+    // Migration: Remove the feature flag and make the uniform dialog the default.
     @if feature-flags.$modal-enable-uniform-dialog {
-        @include -uniform-modal-dialog();
-    }
+        @include -uniform-modal-dialog($from-breakpoint: map.get(theme.$breakpoints, mobile));
+    } @else {
+        @include -docked-modal-dialog();
+        @include -uniform-modal-dialog($from-breakpoint: map.get(theme.$breakpoints, tablet));
 
-    @media (prefers-reduced-motion: no-preference) {
-        transition-property: bottom, width, border-radius, transform; // 1.
-        transition-duration: inherit;
+        [open] & {
```

Comment:
can modals be nested inside each other (and then this would break)?

Reply:
We only support stacking, not nesting. But this scenario sounds so terrifying that I'd rather say it aloud in the README 🙈.

Reply:
OK, I added it there.